### PR TITLE
Requested matches can be played on non-trusted clients

### PR DESCRIPTION
**Pull request by ratosh**

### CODE REVIEW

**Comment by ratosh** [comment]
Working now

File: aiarena/core/management/commands/seed.py [hunk 1]
```diff
@@ -20,6 +20,12 @@ def run_seed(matches, token):
     client.create_arenaclient('aiarenaclient-002', 'aiarenaclient-002@dev.aiarena.net', devadmin.id)
     client.create_user('service_user', 'x', 'service_user@dev.aiarena.net', 'SERVICE', devadmin.id)
 
+    # if token is None it will generate a new one, otherwise it will use the one specified
+    if token is None:
+        api_token = Token.objects.create(user=arenaclient1)
+    else:
+        api_token = Token.objects.create(user=arenaclient1, key=token)
+
```
Comment (lladdy) [inline aiarena/core/management/commands/seed.py:28]: Any specific reason for this change?
I believe it's functionally identical to how it was.

File: aiarena/core/management/commands/seed.py [hunk 2]
```diff
@@ -111,16 +117,14 @@ def run_seed(matches, token):
         CompetitionParticipation.objects.create(competition=competition1, bot=bot)
         CompetitionParticipation.objects.create(competition=competition2, bot=bot)
 
-        # if token is None it will generate a new one, otherwise it will use the one specified
-        api_token = Token.objects.create(user=arenaclient1, key=token)
-        client.set_api_token(api_token)
```
Comment (lladdy) [inline aiarena/core/management/commands/seed.py:116]: This having been removed could potentially be why the seed test is failing.

File: aiarena/core/tests/testing_utils.py [hunk 3]
```diff
@@ -171,11 +167,15 @@ def request_match(self,
         assert response.status_code == 302 and response.url == url
         return Match.objects.filter(requested_by=response.wsgi_request.user).order_by('-created').first()
 
-    def next_match(self) -> Match:
+    def post_to_matches(self) -> Match:
         url = reverse('ac_next_match-list')
-        response = self.django_client.post(url, self.get_api_headers())
+        response = self.django_client.post(url)
```
Comment (lladdy) [inline aiarena/core/tests/testing_utils.py:172]: You've removed the setting of the headers here.
Was this intentional?

File: aiarena/core/management/commands/seed.py (hunk 1 above)
Comment (ratosh) [reply aiarena/core/management/commands/seed.py:28]: Change was part of my try to fix seed testing. I'll revert.

File: aiarena/core/tests/testing_utils.py (hunk 3 above)
Comment (ratosh) [reply aiarena/core/tests/testing_utils.py:172]: Yes

File: aiarena/core/management/commands/seed.py (hunk 2 above)
Comment (ratosh) [reply aiarena/core/management/commands/seed.py:116]: Arena client communication is on a different instance. Will change the new client to behave like the old part using api_token headers only.

**Review by lladdy — CHANGES_REQUESTED** [review]
@ratosh Apologies for the delay in getting around to re-addressing this PR.
 
I'd like to separate out the changes that aren't specifically relevant to requested matches being played by non-trusted clients.
Can you please extract such changes into separate PRs to this one?
Try to minimize non-relevant changes to the existing code in this PR (such as formatting).
This will make it much easier to spot any issues.

I can optionally do it if you'd prefer/don't have time at the moment.

Some quick ones I have identified.
- VS Code related change
- The deletion of exampe-dev-env.py
- Changes to settings.py file
- Docker changes
- Restructure of test code

If some of these things are desirable to keep in this PR, first extract/PR them and I'll merge to staging so the diff between this branch and staging is much simpler.

**Comment by lladdy** [comment]
Closing - PR recreated here: https://github.com/aiarena/aiarena-web/pull/427

**Pull request closed without merging**